src/listpack.c

```diff
@@ -1381,6 +1381,39 @@ void lpRandomPair(unsigned char *lp, unsigned long total_count, listpackEntry *k
     val->sval = lpGetValue(p, &(val->slen), &(val->lval));
 }
 
+/* Randomly select 'count' elements and store pointers to them in the 'results'
+ * array, which needs to have space for 'count' pointers. The order is random
+ * and duplicates are possible. */
+void lpRandomElements(unsigned char *lp, unsigned int count, unsigned char **result) {
```

Comment:
Yes but a listpackEntry is three words instead of one, so I thought it's wasteful. Perhaps refactor the listpackEntry functions towards element pointer instead?

Reply:
the difference is that it saves you the effort to later call lpGet.
are you worried from memory efficiency? listpacks are meant to be small, i don't think this matters.
what could matter is if in some cases you either in some cases you don't need to decode all entries you got (just some of them), so the lpGet is excessive, or if you need the actual pointer in order to delete records and such.

Reply:
Fine, I'll use listpackEntry for this.